plone/app/discussion/catalog.py

```diff
@@ -123,7 +123,7 @@ def created(object):
         object.creation_date.hour,
         object.creation_date.minute,
         object.creation_date.second,
-        "GMT",
+        object.creation_date.tzname(),
```

Comment:
@instification I tested like this:
1. Create a new Plone site with the default timezone (which for me is America/Los_Angeles)
2. Create a page with discussion enabled
3. Add a comment

It broke with this traceback while indexing the comment:
```
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 167, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 376, in publish_module
  Module ZPublisher.WSGIPublisher, line 271, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 68, in call_object
  Module zope.browserpage.simpleviewclass, line 41, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 58, in __call__
  Module zope.pagetemplate.pagetemplate, line 133, in pt_render
  Module Products.PageTemplates.engine, line 365, in __call__
  Module z3c.pt.pagetemplate, line 176, in render
  Module chameleon.zpt.template, line 302, in render
  Module chameleon.template, line 215, in render
  Module chameleon.utils, line 53, in raise_with_traceback
  Module chameleon.template, line 192, in render
  Module b53c3cd94e3d9761bd94c86358cf5869, line 319, in render
  Module 0be9f364dec6f26defecfe5e8acd2b32, line 921, in render_master
  Module 0be9f364dec6f26defecfe5e8acd2b32, line 1608, in render_content
  Module zope.contentprovider.tales, line 76, in __call__
  Module zope.viewlet.manager, line 157, in update
  Module zope.viewlet.manager, line 163, in _updateViewlets
  Module plone.app.discussion.browser.comments, line 320, in update
  Module plone.z3cform.fieldsets.extensible, line 65, in update
  Module plone.z3cform.patch, line 30, in GroupForm_update
  Module z3c.form.group, line 145, in update
  Module plone.app.z3cform.csrf, line 21, in execute
  Module z3c.form.action, line 98, in execute
  Module z3c.form.button, line 301, in __call__
  Module z3c.form.button, line 159, in __call__
  Module plone.app.discussion.browser.comments, line 272, in handleComment
  Module plone.app.discussion.conversation, line 191, in addComment
  Module zope.event, line 32, in notify
  Module zope.component.event, line 27, in dispatch
  Module zope.component._api, line 134, in subscribers
  Module zope.interface.registry, line 448, in subscribers
  Module zope.interface.adapter, line 899, in subscribers
  Module zope.component.event, line 36, in objectEventNotify
  Module zope.component._api, line 134, in subscribers
  Module zope.interface.registry, line 448, in subscribers
  Module zope.interface.adapter, line 899, in subscribers
  Module plone.app.discussion.comment, line 255, in notify_workflow
  Module Products.CMFCore.WorkflowTool, line 300, in notifyCreated
  Module Products.CMFCore.WorkflowTool, line 610, in _reindexWorkflowVariables
  Module Products.CMFCore.CMFCatalogAware, line 110, in reindexObjectSecurity
  Module Products.CMFCore.CatalogTool, line 272, in unrestrictedSearchResults
  Module Products.CMFCore.indexing, line 97, in processQueue
  Module Products.CMFCore.indexing, line 227, in process
  Module Products.CMFCore.indexing, line 49, in reindex
  Module Products.CMFCore.CatalogTool, line 368, in _reindexObject
  Module Products.CMFPlone.CatalogTool, line 324, in catalog_object
  Module Products.ZCatalog.ZCatalog, line 507, in catalog_object
  Module Products.ZCatalog.Catalog, line 341, in catalogObject
  Module Products.ZCatalog.Catalog, line 292, in updateMetadata
  Module Products.ZCatalog.Catalog, line 425, in recordify
  Module plone.indexer.wrapper, line 65, in __getattr__
  Module plone.indexer.delegate, line 20, in __call__
  Module plone.app.discussion.catalog, line 119, in created
  Module DateTime.DateTime, line 447, in __init__
  Module DateTime.DateTime, line 844, in _parse_args
DateTime.interfaces.DateTimeError: DateTime.interfaces.DateTimeError: Unknown time zone: PDT
```

I'd suggest using [plone.app.event.base.DT](https://github.com/plone/plone.app.event/blob/bd412c833364797c001ca74f0d013ae20f2d34c9/plone/app/event/base.py#L565) to convert from datetime to DateTime. I tried it locally and it worked without this error.

Reply:
Weird that the code was happy with CEST but not PDT. When I changed the test timezone to `America/Los_Angeles` the indexers test failed.

So I've implemented your suggestion and changed the test timezone to hopefully improve coverage for timezone issues.